server/app/services/geo/esri/EsriClient.java

```diff
@@ -0,0 +1,88 @@
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_LABELS;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_VALUES;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_URLS;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_PATHS;
+
+  private final WSClient ws;
+
+  @Inject
+  public EsriClient(Config configuration, WSClient ws) {
+    this.ESRI_FIND_ADDRESS_CANDIDATES_URL =
+        checkNotNull(configuration).getString("esri_find_address_candidates_url");
+    this.ESRI_ADDRESS_VERIFICATION_LABELS =
+        checkNotNull(configuration).getList("esri_address_verification_labels");
+    this.ESRI_ADDRESS_VERIFICATION_VALUES =
+        checkNotNull(configuration).getList("esri_address_verification_values");
+    this.ESRI_ADDRESS_VERIFICATION_URLS =
+        checkNotNull(configuration).getList("esri_address_verification_urls");
+    this.ESRI_ADDRESS_VERIFICATION_PATHS =
+        checkNotNull(configuration).getList("esri_address_verification_paths");
+    this.ws = ws;
+  }
+
+  public CompletionStage<JsonNode> getAddressCandidates(AddressAttributes addressAttributes) {
+    WSRequest request = ws.url(this.ESRI_FIND_ADDRESS_CANDIDATES_URL);
+    request.setContentType(ESRI_CONTENT_TYPE);
+    request.addQueryParameter("outFields", ESRI_FIND_ADDRESS_CANDIDATES_OUT_FIELDS);
+    request.addQueryParameter("f", ESRI_FIND_ADDRESS_CANDIDATES_FORMAT);
+    // add addressAttributes to query
+    request.addQueryParameter("address", addressAttributes.getStreetValue());
+    request.addQueryParameter("address2", addressAttributes.getLine2Value());
+    request.addQueryParameter("city", addressAttributes.getCityValue());
+    request.addQueryParameter("region", addressAttributes.getStateValue());
+    request.addQueryParameter("postal", addressAttributes.getZipValue());
+    CompletionStage<JsonNode> jsonFuture = request.get().thenApply(res -> res.getBody(json()));
+    return jsonFuture;
+  }
+
+  public CompletionStage<JsonNode> verifyAddressCoordinates(
+      String verificationValue, AddressLocation location) {
+    List<String> verificationValues =
+        this.ESRI_ADDRESS_VERIFICATION_VALUES.stream()
+            .map(configValue -> (String) configValue.unwrapped())
+            .collect(Collectors.toList());
+
+    int verificationIndex = verificationValues.indexOf(verificationValue);
+
+    String verificationUrl =
+        this.ESRI_ADDRESS_VERIFICATION_URLS.get(verificationIndex).unwrapped().toString();
+
+    WSRequest request = ws.url(verificationUrl);
+    request.setContentType(ESRI_CONTENT_TYPE);
+    request.addQueryParameter("f", ESRI_FIND_ADDRESS_CANDIDATES_FORMAT);
+    request.addQueryParameter("geometryType", "esriGeometryPoint");
+    request.addQueryParameter("outFields", "*");
+    String geo = "{'x':";
+    geo += location.getX();
+    geo += ",'y':";
+    geo += location.getY();
+    geo += ",'spatialReference':";
+    geo += location.getWkid();
+    request.addQueryParameter("geometry", geo);
```

Comment:
suggestion (blocking): Just eyeballing it, it looks like this JSON string is missing a closing curly brace. We should probably use a library for building JSON documents and then serializing them to strings, maybe `JsonNode`. A quicker fix since we're moving fast could be to use a format string here instead
```suggestion
    String geo = String.format(
      "{\"x\": %d, \"y\": %d, \"spatialReference\": \"%s\"}",
      location.getLongitude(),
      location.getLatitude(),
      location.getWellKnownId()
    );
    request.addQueryParameter("geometry", geo);
```

Need a unit test that parses the JSON to ensure it's valid.